lib/ts/station/fsm.ex

```diff
@@ -0,0 +1,205 @@
+defmodule Station.Fsm do
+  @moduledoc """
+  Provides implementation of the core logic of a Station.
+  """
+
+  use Fsm, initial_state: :start, initial_data: []
+  @behaviour Station.FsmBehaviour
+
+  # Module interface definition
+  def initialise_fsm(input = [_station_struct, _dependency]) do
+    new()
+    |> input_data(input)
+  end
+
+  def update_station(station_fsm, new_station_struct) do
+    station_fsm
+    |> update(new_station_struct)
+  end
+
+  def process_itinerary(station_fsm, itinerary) do
```

Comment:
please remind me the bug (a missing module API call) we found during the code review and your fix for the same.